airflow/utils/log/file_task_handler.py

```diff
@@ -218,9 +242,12 @@ def close(self):
         if self.handler:
             self.handler.close()
 
-    def _render_filename(self, ti: TaskInstance, try_number: int) -> str:
+    def _render_filename(self, ti: TaskInstance | TaskInstanceKey, try_number: int) -> str:
         """Returns the worker log filename."""
+        from airflow.models.taskinstance import TaskInstance
+
         with create_session() as session:
+            ti = TaskInstance.from_ti(ti, session=session)
             dag_run = ti.get_dagrun(session=session)
```

Comment:
similar to above, I suspect that there is an optimization here, if USE_PER_RUN_LOG_ID  is false, then we should not need to go to the database i think.  this is probably not a huge issue on individual task logging i.e. on the worker but from something like scheduler you would probably want to avoid queries.  though, not sure if it's really possible given the templating etc.

Reply:
`USE_PER_RUN_LOG_ID` is not used here, and the case that `USE_PER_RUN_LOG_ID == False` is when users use airflow < 2.3.3. As I think we don't want to roll back the code to 2.3.3, I'm confused about what you mean by `USE_PER_RUN_LOG_ID is false` here.

Reply:
Yeah as you mentioned on slack, i was just making quick conclusions based on what i saw in the code but it appears that this is not a global feature, just a backcompat flag in ES handler. but maybe we should make it a config and then we could use it here.